test/suites/drift/suite_test.go

```diff
@@ -148,20 +150,77 @@ var _ = Describe("Drift", Label("AWS"), func() {
 			g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(node), &v1.Node{})).To(Succeed())
 		}).WithTimeout(time.Minute).Should(Succeed())
 	})
-})
+	It("should deprovision nodes that have drifted due to securitygroup", func() {
+		By("getting the cluster vpc id")
+		output, err := env.EKSAPI.DescribeCluster(&eks.DescribeClusterInput{Name: awssdk.String(settings.FromContext(env.Context).ClusterName)})
+		Expect(err).To(BeNil())
+
+		By("creating new security group")
+		createSecurityGroup := &ec2.CreateSecurityGroupInput{
+			GroupName:   awssdk.String("security-group-drift"),
+			Description: awssdk.String("End-to-end Drift Test, should delete after drift test is completed"),
+			VpcId:       output.Cluster.ResourcesVpcConfig.VpcId,
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: awssdk.String("security-group"),
+					Tags: []*ec2.Tag{
+						{
+							Key:   awssdk.String("karpenter.sh/discovery"),
+							Value: awssdk.String(settings.FromContext(env.Context).ClusterName),
+						},
+					},
+				},
+			},
+		}
+		_, _ = env.EC2API.CreateSecurityGroup(createSecurityGroup)
+
+		By("looking for security groups")
+		securitygroups := env.GetSecurityGroups(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		testSecurityGroup, _ := lo.Find(securitygroups, func(sg aws.SecurityGroup) bool {
+			return awssdk.StringValue(sg.GroupName) == "security-group-drift"
+		})
```

Comment:
Can we do some sort of eventually here? I can imagine since we just created this that there could be some races.

Reply:
This is a good callout. Does `CreateSecurityGroup` not just give us the securityGroup back if it's already created or is it only an error. If it's the latter, then I think the `Eventually` makes sense.

Reply:
`CreateSecurityGroup` only provides the SG id

Reply:
`CreateSecurityGroup` does not give us back an ID if it's already created. That's why don't think an `Eventually` will work here